src/plugins/expressions/common/expression_functions/specs/kibana_context.ts

```diff
@@ -74,13 +73,21 @@ export const kibanaContext = (): ExpressionFunctionKibanaContext => ({
       }),
     },
   },
-  async fn(input, args) {
+
+  async fn(input, args, { getSavedObject }) {
     const queryArg = args.q ? JSON.parse(args.q) : [];
     let queries = Array.isArray(queryArg) ? queryArg : [queryArg];
     let filters = args.filters ? JSON.parse(args.filters) : [];
 
     if (args.savedSearchId) {
-      const obj = await savedObjects.get('search', args.savedSearchId);
+      if (typeof getSavedObject !== 'function') {
+        throw new Error(
+          '"getSavedObject" function not available in execution context. ' +
+            'When you execute expression you need to extra execution context ' +
```

Comment:
```suggestion
            'When you execute expression you need to add extra execution context ' +
```

Reply:
if we use the getter/setter approach it will already handle this error